src/main/kotlin/com/dto/ReservationManager.kt

```diff
@@ -0,0 +1,61 @@
+package com.dto
+
+import com.excpetion.ReservationAlreadyStartedException
+import com.excpetion.ReservationNotFoundException
+import com.model.ReservationDao
+import com.parkingrate.ParkingRateEvaluator
+import com.util.ReservationCalculator
+import org.springframework.stereotype.Service
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.Optional
+import javax.transaction.Transactional
+
+
+@Service
+class ReservationManager(private val reservationDao: ReservationDao,
+                         private val reservationConverter: ReservationConverter,
+                         private val reservationCalculator: ReservationCalculator) {
+
+    @Transactional
+    fun createReservation(reservationRequest: ReservationRequest, startTime: LocalDateTime): StartedReservationDto {
+
+        Optional.ofNullable(reservationDao.findStartedReservation(reservationRequest.carLicensePlate))
```

Comment:
skoro to kotlin to po co korzystasz z Optionala? Kotlin ma do tego inne mechanizmy

Reply:
Tak to prawda, ale szukając sposobów kotlinowskich doszedłem do wniosku że użycie Optionala z ifPresent było najczytelniejsze.